### ISSUE TITLE
Grampshub: no sync from desktop to web | no edit on web 

### ISSUE THREAD

**Issue by petrmint** [gramps-project/gramps-web-api#507]
I am fairly new to the process so I may have gone wrong at some point but:

- I opted for the Grampshub solution, followed the advice (incl. database integrity check prior to exporting xml from the desktop), and successfully loaded the tree (1919 persons) to the Grampshub web. Then I added 154 media files (after some retries) and all seemed fine.
- I installed the web sync add-on to my desktop
- Then after changing couple of things in my tree on the desktop, I tried to sync the changes to web. The add-on says it went OK but in fact, nothing got synced.

- Moreover, I cannot seem to edit anything on the web either - cannot add objects, cannot edit, cannot delete. I even tried to delete all the objects ("Danger Zone") but it throws an error "??? Unknown error: none" and nothing gets deleted.

This is the system info copied from Grampshub:
Gramps 5.2.0
Gramps Web API 2.1.0
Gramps Web Frontend 24.3.1
locale: en
multi-tree: true
task queue: true

I am the owner of the tree (I cannot seem to add an admin account from the web either - "Not authorised"; I thought I might lack some  authorisation in the owner role... 🤷‍♂️)

BTW, my tree of cca 1900 persons originated as a Gedcom export from MyHeritage that I loaded onto Gramps 5.2 - so there might be some conversion issues hidden there since I started the tree years ago without any knowledge and it gradually collected some MH garbage that I wanted to cleanup in Gramps :-)

**Comment by petrmint** [comment]
Thanks for coming back.

> Indeed, you're the owner of the tree, so your role is "owner". The "administrator" role has rights to administer trees other than their own, which you don't have.

Aha; got it.

My edit / delete / new note attempts while logged in as an owner (via Edge browser):
* with F12, trying to edit an existing note on the web (that was loaded as part of the original tree via gramps xml), after saving the change, I can see an error in Console section "PUT https://grampshub.com/api/notes/f8dab849ebb3a02f98bc5d659ea 500 (Internal Server Error)" - not sure what to copy, sorry: das ist mir ein böhmisches Dorf ;-)
![image](https://github.com/gramps-project/gramps-web-api/assets/164870327/1af3c23b-0919-4dc8-82bc-2101c7275c2b)

* with F12, trying to "Delete all objects", it would not do anything visible, but it says this in the DevTools section:
![image](https://github.com/gramps-project/gramps-web-api/assets/164870327/126fdedc-2bba-4165-9fc5-6e8da5f17f07)

* with F12 on, a "new note" attempt throws a "POST https://grampshub.com/api/notes/ 500 (Internal Server Error)"
![image](https://github.com/gramps-project/gramps-web-api/assets/164870327/30fa52fd-cb01-45ec-ac22-7abf2ba95dfa)
For a moment, the main window also showed a comment "Error: the Gramps database needs a schema upgrade"
![image](https://github.com/gramps-project/gramps-web-api/assets/164870327/5bf00f77-2418-4f4a-a471-ba3be27dbc79)



As mentioned, I can see the following in the system info section (web):
_Gramps 5.2.0
Gramps Web API 2.1.0
Gramps Web Frontend 24.3.1
locale: en
multi-tree: true
task queue: true_

desktop version is this
_GRAMPS: AIO64-5.2.0-r1-6167151 
Python: 3.11.8 
BSDDB: 6.2.9 (6, 0, 30) 
sqlite: 3.45.1 (2.6.0)
LANG: cs_CZ.UTF-8
OS: Windows_


thanks and best regards
pm

**Comment by DavidMStraub** [comment]
Hi,

I see, your database is not converted to the Gramps 5.2 schema yet, see here
https://www.grampshub.com/blog/en/2024/03/02/gramps-52-and-grampshub.html (en)
https://www.grampshub.com/blog/de/2024/03/02/gramps-52-und-grampshub.html (de)

However in that case you should be shown with the upgrade screen and nothing else. I'm puzzled why that's not the case. Can you please logout and login again? And possible try with a different browser?

**Comment by petrmint** [comment]
Logged out from Edge and re-logged in from Chrome but still no upgrade screen (I can normally access the tree as loaded during the first import, but any editing is not possible)

Timeline backround, perhaps it makes a difference:
Unaware of the fact Gramps project was going through such a major change (I mean shift from 5.1 >>> 5.2), I installed Gramps literally on 24-Feb-2024 :-) i.e. one day after the shift has happened - so I installed the 5.2.0 version already, the net effect being I never had to convert anything in the desktop version as I started with an empty Gramps 5.2 first place; I never had any part of the tree in 5.1 format. 
So on 24-Feb, I loaded my (admittedly messed-up) GEDCOM from MyHeritage into Gramps 5.2, and populated my tree in Gramps 5.2 from that source. Then I was playing with the desktop version a bit, doing some (very little) cleanup of the data (repositories, notes, citations), and after some time, only this very weekend (23rd/24th March) I tried to export the tree as a XML out of Gramps 5.2 desktop. 
In other words, I believe I started with Gramps 5.2 desktop, exported the tree on 23rd March into XML (I kept the "compressed" switch on - it just wraps it into ZIP, right?), and loaded it into Grampshub >>> so it happened after 9th of March that I understand was the date Grampshub underwent the upgrade to 5.2 as well.

But what you say is that at some point, the underlying database somehow started to think it is in Gramps 5.1 format? I am not aware of any user action that I would knowingly trigger like "downgrade database back to 5.1"... 🤔

pm

**Comment by DavidMStraub** [comment]
Thanks for checking and for the background! I will investigate in a few hours and get back to you.

**Comment by DavidMStraub** [comment]
I'm puzzled.

Can you please open the network tab in the browser dev tools and share what you are seeing for database/actual_schema and database/schema?

![image](https://github.com/gramps-project/gramps-web-api/assets/10965193/f9d9b09e-142e-4bc5-9107-33f65600ac67)

**Comment by petrmint** [comment]
in Chrome, it says this (I replaced "id:" with xxxx because, I dunno, you replaced it in your picture too 😁)

    "database": {
        "actual_schema": 0,
        "id": "xxxxxxxxxxxxxxxxxx",
        "name": "My Tree",
        "schema": "20.0.0",
        "type": "Unknown",
        "version": "2.9.5 (dt dec pq3 ext lo64)"

**Comment by DavidMStraub** [comment]
OK, actual schema shouldn't be 0, I will look into it. Can you please share the "id" with me via e-mail or Grampshub ticket? Then I can investigate.

**Comment by petrmint** [comment]
Hi, 

this is the ID: b5ed2aee-fb64-4df4-ae8f-b2e21e25b0e6
   

"database": {
        "actual_schema": 0,
        "id": "b5ed2aee-fb64-4df4-ae8f-b2e21e25b0e6",
        "name": "My Tree",
        "schema": "20.0.0",
        "type": "Unknown",
        "version": "2.9.5 (dt dec pq3 ext lo64)"
 
thanks a lot for investigating
pm
______________________________________________________________
> Od: "David Straub" ***@***.***>
> Komu: "gramps-project/gramps-web-api" ***@***.***>
> Datum: 27.03.2024 12:00
> Předmět: Re: [gramps-project/gramps-web-api] Grampshub: no sync from desktop to web | no edit on web (Issue #507)
>
 
OK, actual schema shouldn't be 0, I will look into it. Can you please share the "id" with me via e-mail or Grampshub ticket? Then I can investigate.

**Comment by DavidMStraub** [comment]
@petrmint please try now.

**Comment by petrmint** [comment]
Thanks a lot: the web editing functions do work now (I am travelling for Easter so I will be able to test the sync with desktop only next week)
 
best regards
pm
 
______________________________________________________________
> Od: "David Straub" ***@***.***>
> Komu: "gramps-project/gramps-web-api" ***@***.***>
> Datum: 27.03.2024 21:22
> Předmět: Re: [gramps-project/gramps-web-api] Grampshub: no sync from desktop to web | no edit on web (Issue #507)
>
 
@petrmint <https://github.com/petrmint> please try now.

**Comment by petrmint** [comment]
The sync is working also for desktop now, both ways.
Thanks a lot for help.
pm
